Move to latest NuGet.exe

dougbu opened the pull request:
- 5.3.0 -> 5.6.0
- should improve performance and may improve reliability

dougbu committed:
Move to latest NuGet.exe
- 5.3.0 -> 5.6.0
- should improve performance and may improve reliability

wtgodbe commented:
What about other references to nuget in the repo? There's at least one other:

https://github.com/dotnet/aspnetcore/blob/bd917f690d1340d0155ae2fcf0e5ef731ebf486c/eng/tools/RepoTasks/RepoTasks.csproj#L15

dougbu commented on src/Installers/Windows/GenerateNugetPackageWithMsi.ps1:
```diff
@@ -25,9 +25,8 @@ if (-not (Test-Path $NuGetDir)) {
 }
 
 if (-not (Test-Path $NuGetExe)) {
-    # Using 5.3.0 to workaround https://github.com/NuGet/Home/issues/5016
```
@rrelyea this bug doesn't appear relevant anymore, especially because we aren't using the more-performant 3.5.0 version. If that's correct, I'm wondering if is `dotnet pack` able to handle https://github.com/dotnet/aspnetcore/blob/master/src/Installers/Windows/SharedFramework/SharedFrameworkPackage.nuspec and https://github.com/dotnet/aspnetcore/blob/master/src/Installers/Windows/TargetingPack/TargetingPackPackage.nuspec as well as the properties we pass in below yet ❔ I don't see `*.nuspec`support in the help `dotnet pack -?` spits out but hope it's just undocumented 😄 or will be coming soon.

dougbu replied:
Chatted offline w/ @rrelyea and it's likely we could create the two installer packages using a new `*.csproj` that runs after our `*.wixproj`.

Will file a separate issue about this work. It's a larger change and not something I'm going to tackle now. But, it would remove our need for NuGet.exe anywhere in this repo (outside a couple of Arcade post-build stages).

dougbu commented:
> What about other references to nuget in the repo? There's at least one other:

> https://github.com/dotnet/aspnetcore/blob/bd917f690d1340d0155ae2fcf0e5ef731ebf486c/eng/tools/RepoTasks/RepoTasks.csproj#L15

@rrelyea would changing that one negatively impact loading our tasks into `dotnet msbuild` or `msbuild`❔ If not, I'll add another commit&hellip;

dougbu commented:
And @rrelyea if I shouldn't be asking you NuGet questions anymore please add the right contact.

rrelyea commented:
it might be easier to have a quick teams call to chat through these several questions. call me sometime?

dougbu commented:
Sounds good @rrelyea

dougbu commented:
@rrelyea and I also discussed updating https://github.com/dotnet/aspnetcore/blob/bd917f690d1340d0155ae2fcf0e5ef731ebf486c/eng/tools/RepoTasks/RepoTasks.csproj#L15 There should be no downside to moving to 5.6.0. @rrelyea also pointed out we don't need all of NuGet.Build.Tasks and could move to getting just NuGet.Packaging. Overall, that transitively brings in 5 packages instead of the current 12. I confirmed this looking at RepoTasks.dll and am going to try this narrowing out in the next iteration.

dougbu committed:
Also switch from NuGet.Build.Tasks to NuGet.Packaging
- move to 5.6.0 version here too
- reduce transitive dependencies; we don't need them all

dougbu commented:
/ping @dotnet/aspnet-build this is building successfully and just needs a review 😺
